Sources/SwiftProtobuf/DelimitedMessageUtil.swift

```diff
@@ -0,0 +1,215 @@
+// Sources/SwiftProtobuf/DelimitedMessageUtil.swift - Delimited support
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/master/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+///
+/// Helpers to read/write message with a length prefix.
+///
+// -----------------------------------------------------------------------------
+
+import Foundation
+
+/// Additional errors for delimited message handing.
+public enum DelimitedMessageError: Error {
+  /// If a read/write to the stream fails, but the stream's `streamError` is nil,
+  /// this error will be throw instead since the stream didn't provide anything
+  /// more specific. A common cause for this can be failing to open the stream
+  /// before trying to read/write to it.
+  case unknownStreamError
+
+  /// While reading/writing to the stream, less than the expected bytes was
+  /// read/written.
+  case truncated
+}
+
+/// Helper methods for reading/writing messages with a length prefix.
+public enum DelimitedMessageUtil {
+
+  /// Write a single size-delimited message from the given stream. Delimited
+  /// format allows a single file or stream to contain multiple messages,
+  /// whereas normally writing multiple non-delimited messages to the same
+  /// stream would cause them to be merged. A delimited message is a varint
+  /// encoding the message size followed by a message of exactly that size.
+  ///
+  /// - Parameters:
+  ///   - message: The message to be written.
+  ///   - to: The `OutputStream` to write the message too.  The stream is
+  ///     is assumed to be ready to be written too.
+  ///   - partial: If `false` (the default), this method will check
+  ///     `Message.isInitialized` before encoding to verify that all required
+  ///     fields are present. If any are missing, this method throws
+  ///     `BinaryEncodingError.missingRequiredFields`.
+  /// - Throws: `BinaryEncodingError` if encoding fails; throws
+  ///           `DelimitedMessageError` for some writing errors, and the
+  ///           underlying OutputStream.streamError for a stream error.
+  public static func write(
```

Comment:
Hm, I was focused on the write/parse side of things, but yea, maybe we need to capture "serialize" in here to better capture which format it is.  We could also capture this in the helper type - `DelimitedBinarySerization` ?